File: Problem-1710/Solution.c
#include <stdlib.h>

int cmp(const void *a, const void *b) {
    int *boxA = *(int **)a;
    int *boxB = *(int **)b;
    return boxB[1] - boxA[1];  
}

int maximumUnits(int** boxTypes, int boxTypesSize, int* boxTypesColSize, int truckSize) {
    qsort(boxTypes, boxTypesSize, sizeof(int*), cmp);

    int totalUnits = 0;

    for (int i = 0; i < boxTypesSize && truckSize > 0; i++) {
        int numBoxes = boxTypes[i][0];
        int unitsPerBox = boxTypes[i][1];

        int take = (numBoxes < truckSize) ? numBoxes : truckSize;

        totalUnits += take * unitsPerBox;
        truckSize -= take;
    }

    return totalUnits;
}

File: Problem-1710/Solution.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a,b) -> b[1] - a[1]);
        int ans = 0;

        for(int i = 0; i < boxTypes.length && truckSize > 0; i++){
            int carry = Math.min(boxTypes[i][0], truckSize);

            ans = ans + (carry * boxTypes[i][1]);
            truckSize = truckSize - carry;
        }
        return ans;
    }
}
